File: scripts/ai-proxy-test/test-highwayapi.mjs
// Test de connectivité vers le proxy tiers HighwayAPI (api.highwayapi.ai).
// ATTENTION: ce n'est PAS l'API Anthropic officielle — ne pas utiliser en
// production avec des données d'élèves/parents tant que ce proxy n'a pas
// été validé (confidentialité, fiabilité). Voir data_import/ai_parser.py
// pour l'intégration officielle (SDK anthropic + ANTHROPIC_API_KEY).
import OpenAI from "openai";

const apiKey = process.env.HIGHWAY_API_KEY;
if (!apiKey) {
  console.error("Définir HIGHWAY_API_KEY dans l'environnement avant de lancer ce script.");
  process.exit(1);
}

const client = new OpenAI({
  baseURL: "https://api.highwayapi.ai/openai",
  apiKey,
});

const stream = true;

async function run() {
  const completion = await client.chat.completions.create({
    messages: [
      { role: "system", content: "Tu es un assistant utile." },
      { role: "user", content: "Hi there!" },
    ],
    model: "claude-sonnet-5",
    stream,
  });

  if (stream) {
    for await (const chunk of completion) {
      const delta = chunk.choices[0].delta.content;
      if (delta) process.stdout.write(delta);
      if (chunk.choices[0].finish_reason) {
        console.log(`\n[finish_reason: ${chunk.choices[0].finish_reason}]`);
      }
    }
  } else {
    console.log(JSON.stringify(completion, null, 2));
  }
}

run().catch((err) => {
  console.error("Échec de l'appel:", err.message ?? err);
  process.exit(1);
});

File: scripts/ai-proxy-test/test_highwayapi.py
"""Test de connectivite vers le proxy tiers HighwayAPI (api.highwayapi.ai).

ATTENTION: ce n'est PAS l'API Anthropic officielle - ne pas utiliser en
production avec des donnees d'eleves/parents tant que ce proxy n'a pas ete
valide (confidentialite, fiabilite). Voir data_import/ai_parser.py pour
l'integration officielle (SDK anthropic + ANTHROPIC_API_KEY).

Usage:
    export HIGHWAY_API_KEY="..."
    python test_highwayapi.py
"""
import os
import sys

from openai import OpenAI

api_key = os.environ.get("HIGHWAY_API_KEY")
if not api_key:
    print("Definir HIGHWAY_API_KEY dans l'environnement avant de lancer ce script.", file=sys.stderr)
    sys.exit(1)

client = OpenAI(base_url="https://api.highwayapi.ai/openai", api_key=api_key)

stream = True

response = client.chat.completions.create(
    model="claude-sonnet-5",
    messages=[
        {"role": "system", "content": "Tu es un assistant utile."},
        {"role": "user", "content": "Hi there!"},
    ],
    stream=stream,
    max_tokens=1024,
    temperature=1,
    presence_penalty=0,
    frequency_penalty=0,
    response_format={"type": "text"},
    extra_body={"top_k": 50, "min_p": 0},
)

if stream:
    for chunk in response:
        if not chunk.choices:
            continue
        delta = chunk.choices[0].delta.content
        if delta:
            print(delta, end="", flush=True)
    print()
else:
    print(response.choices[0].message.content)
